diff --git a/run_tests.sh b/run_tests.sh
index e48f1db..4285b52 100755
--- a/run_tests.sh
+++ b/run_tests.sh
@@ -29,7 +29,7 @@ function run()
     fi
 
     if [ -n "$arch" ] && [ "$arch" != "$ARCH" ]; then
-        echo "skip $1 ($arch only)"
+        #echo "skip $1 ($arch only)"
         return
     fi
 
@@ -40,25 +40,25 @@ function run()
         path=${check_param%%=*}
         value=${check_param#*=}
         if [ "$path" ] && [ "$(cat $path)" != "$value" ]; then
-            echo "skip $1 ($path not equal to $value)"
+            #echo "skip $1 ($path not equal to $value)"
             return
         fi
     done

     cmdline="./$TEST_DIR-run $kernel -smp $smp $opts"
     if [ $verbose != 0 ]; then
-        echo $cmdline
+        echo $testname : $cmdline
     fi
 
     # extra_params in the config file may contain backticks that need to be
     # expanded, so use eval to start qemu
-    eval $cmdline >> test.log
+    #eval $cmdline >> test.log
 
-    if [ $? -le 1 ]; then
-        echo -e "\e[32mPASS\e[0m $1"
-    else
-        echo -e "\e[31mFAIL\e[0m $1"
-    fi
+    #if [ $? -le 1 ]; then
+    #    echo -e "\e[32mPASS\e[0m $1"
+    #else
+    #    echo -e "\e[31mFAIL\e[0m $1"
+    #fi
 }
 
 function run_all()
