From 6d1d2d2402bede1402ec08f791ec319ffccd8198 Mon Sep 17 00:00:00 2001
From: Brad Figg <brad.figg@canonical.com>
Date: Tue, 5 Dec 2017 07:52:45 -0800
Subject: [PATCH] runtime.bash: Remove the color coding of SKIP/PASS/FAIL

Signed-off-by: Brad Figg <brad.figg@canonical.com>
---
 scripts/runtime.bash | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/runtime.bash b/scripts/runtime.bash
index a31ae91..59a9bb9 100644
--- a/scripts/runtime.bash
+++ b/scripts/runtime.bash
@@ -2,9 +2,9 @@
 : ${MAX_SMP:=$(getconf _NPROCESSORS_CONF)}
 : ${TIMEOUT:=90s}
 
-PASS() { echo -ne "\e[32mPASS\e[0m"; }
-SKIP() { echo -ne "\e[33mSKIP\e[0m"; }
-FAIL() { echo -ne "\e[31mFAIL\e[0m"; }
+PASS() { echo -ne "PASS"; }
+SKIP() { echo -ne "SKIP"; }
+FAIL() { echo -ne "FAIL"; }
 
 extract_summary()
 {
-- 
2.14.1

