diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index d78fe0e..fa152ae 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -1,3 +1,7 @@
+# UBUNTU:
+# This will be copied over the upstream kvm-unit-test; need to periodically
+# check if this is in sync when snapshotting.
+
 # Define your new unittest following the convention:
 # [unittest_name]
 # file = foo.flat # Name of the flat file to be used
@@ -6,11 +10,11 @@
 # arch = i386/x86_64 # Only if the test case works only on one of them
 # groups = group1 group2 # Used to identify test cases with run_tests -g ...
 
-[apic]
-file = apic.flat
-smp = 2
-extra_params = -cpu qemu64,+x2apic
-arch = x86_64
+#[apic]
+#file = apic.flat
+#smp = 2
+#extra_params = -cpu qemu64,+x2apic
+#arch = x86_64
 
 [smptest]
 file = smptest.flat
@@ -69,9 +73,9 @@ arch = x86_64
 #[asyncpf]
 #file = asyncpf.flat
 
-[emulator]
-file = emulator.flat
-arch = x86_64
+#[emulator]
+#file = emulator.flat
+#arch = x86_64
 
 [eventinj]
 file = eventinj.flat
@@ -86,11 +90,11 @@ arch = x86_64
 #[init]
 #file = init.flat
 
-[msr]
-file = msr.flat
+#[msr]
+#file = msr.flat
 
-[pmu]
-file = pmu.flat
+#[pmu]
+#file = pmu.flat
 
 [port80]
 file = port80.flat
@@ -98,8 +102,8 @@ file = port80.flat
 [realmode]
 file = realmode.flat
 
-[s3]
-file = s3.flat
+#[s3]
+#file = s3.flat
 
 [sieve]
 file = sieve.flat
@@ -109,6 +113,7 @@ file = tsc.flat
 
 [tsc_adjust]
 file = tsc_adjust.flat
+extra_params = -cpu host
 
 [xsave]
 file = xsave.flat
@@ -145,16 +150,16 @@ file = kvmclock_test.flat
 smp = 2
 extra_params = --append "10000000 `date +%s`"
 
-[pcid]
-file = pcid.flat
-extra_params = -cpu qemu64,+pcid
-arch = x86_64
+#[pcid]
+#file = pcid.flat
+#extra_params = -cpu qemu64,+pcid
+#arch = x86_64
 
-[vmx]
-file = vmx.flat
-extra_params = -cpu host,+vmx
-arch = x86_64
+#[vmx]
+#file = vmx.flat
+#extra_params = -cpu host,+vmx
+#arch = x86_64
 
-[debug]
-file = debug.flat
-arch = x86_64
+#[debug]
+#file = debug.flat
+#arch = x86_64
