From 7a20801d7fca9bde219d85059fb748a56abf68a8 Mon Sep 17 00:00:00 2001
From: Tim Gardner <tim.gardner@canonical.com>
Date: Thu, 16 Feb 2017 06:40:56 -0700
Subject: [PATCH 1/2] UBUNTU: selftests: just build bpf

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 tools/testing/selftests/Makefile | 39 +++------------------------------------
 1 file changed, 3 insertions(+), 36 deletions(-)

diff --git a/tools/testing/selftests/Makefile b/tools/testing/selftests/Makefile
index 831022b..e4422ff 100644
--- a/tools/testing/selftests/Makefile
+++ b/tools/testing/selftests/Makefile
@@ -1,40 +1,7 @@
+#
+# We only are interested in bpf.
+#
 TARGETS =  bpf
-TARGETS += breakpoints
-TARGETS += capabilities
-TARGETS += cpu-hotplug
-TARGETS += efivarfs
-TARGETS += exec
-TARGETS += firmware
-TARGETS += ftrace
-TARGETS += futex
-TARGETS += gpio
-TARGETS += ipc
-TARGETS += kcmp
-TARGETS += lib
-TARGETS += membarrier
-TARGETS += memfd
-TARGETS += memory-hotplug
-TARGETS += mount
-TARGETS += mqueue
-TARGETS += net
-TARGETS += nsfs
-TARGETS += powerpc
-TARGETS += pstore
-TARGETS += ptrace
-TARGETS += seccomp
-TARGETS += sigaltstack
-TARGETS += size
-TARGETS += static_keys
-TARGETS += sync
-TARGETS += sysctl
-ifneq (1, $(quicktest))
-TARGETS += timers
-endif
-TARGETS += user
-TARGETS += vm
-TARGETS += x86
-TARGETS += zram
-#Please keep the TARGETS list alphabetically sorted
 # Run "make quicktest=1 run_tests" or
 # "make quicktest=1 kselftest from top level Makefile
 
-- 
2.7.4

