From 1fc37df7738c6aeaee6934a6c374a7ef3836cd0b Mon Sep 17 00:00:00 2001
From: Brad Figg <brad.figg@canonical.com>
Date: Wed, 15 Mar 2017 10:36:41 -0700
Subject: [PATCH] UBUNTU: selftests: just build bpf

Signed-off-by: Brad Figg <brad.figg@canonical.com>
---
 tools/testing/selftests/Makefile | 35 -----------------------------------
 1 file changed, 35 deletions(-)

diff --git a/tools/testing/selftests/Makefile b/tools/testing/selftests/Makefile
index 9135188..fd3ca00 100644
--- a/tools/testing/selftests/Makefile
+++ b/tools/testing/selftests/Makefile
@@ -1,39 +1,4 @@
 TARGETS =  bpf
-TARGETS += breakpoints
-TARGETS += capabilities
-TARGETS += cpu-hotplug
-TARGETS += efivarfs
-TARGETS += exec
-TARGETS += firmware
-TARGETS += ftrace
-TARGETS += futex
-TARGETS += gpio
-TARGETS += ipc
-TARGETS += kcmp
-TARGETS += lib
-TARGETS += membarrier
-TARGETS += memfd
-TARGETS += memory-hotplug
-TARGETS += mount
-TARGETS += mqueue
-TARGETS += net
-TARGETS += nsfs
-TARGETS += powerpc
-TARGETS += pstore
-TARGETS += ptrace
-TARGETS += seccomp
-TARGETS += sigaltstack
-TARGETS += size
-TARGETS += static_keys
-TARGETS += sync
-TARGETS += sysctl
-#ifneq (1, $(quicktest))
-#TARGETS += timers
-#endif
-TARGETS += user
-TARGETS += vm
-TARGETS += x86
-TARGETS += zram
 #Please keep the TARGETS list alphabetically sorted
 # Run "make quicktest=1 run_tests" or
 # "make quicktest=1 kselftest from top level Makefile
-- 
2.7.4

