diff --git a/lib/tlibio.c b/lib/tlibio.c
index f2078da..9b1b890 100644
--- a/lib/tlibio.c
+++ b/lib/tlibio.c
@@ -209,7 +209,7 @@ int	*max;
  * This function will allow someone to set the debug level.
  ***********************************************************************/
 int
-lio_set_debug(level)
+lio_set_debug(int level)
 {
     int old;
 
diff --git a/ltp/doio.c b/ltp/doio.c
index 966cff1..7c3d2c7 100644
--- a/ltp/doio.c
+++ b/ltp/doio.c
@@ -36,6 +36,7 @@
 #include <sys/wait.h>
 #include <sys/time.h>	/* for delays */
 #include <ctype.h>
+#include <stdarg.h>
 
 struct io_req;
 int do_xfsctl(struct io_req *);
diff --git a/ltp/fsx.c b/ltp/fsx.c
index 0bb8ae8..f261cd0 100644
--- a/ltp/fsx.c
+++ b/ltp/fsx.c
@@ -207,9 +207,9 @@ prt(char *fmt, ...)
 	va_start(args, fmt);
 	vsnprintf(buffer, BUF_SIZE, fmt, args);
 	va_end(args);
-	fprintf(stdout, buffer);
+	fprintf(stdout, "%s", buffer);
 	if (fsxlogf)
-		fprintf(fsxlogf, buffer);
+		fprintf(fsxlogf, "%s", buffer);
 }
 
 void
diff --git a/src/bulkstat_unlink_test.c b/src/bulkstat_unlink_test.c
index afd69ff..0269a3e 100644
--- a/src/bulkstat_unlink_test.c
+++ b/src/bulkstat_unlink_test.c
@@ -9,6 +9,9 @@
 #include <xfs/xfs.h>
 #include <unistd.h>
 #include <getopt.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
 
 int
 main(int argc, char *argv[])
diff --git a/src/bulkstat_unlink_test_modified.c b/src/bulkstat_unlink_test_modified.c
index 3a50d61..608ecc5 100644
--- a/src/bulkstat_unlink_test_modified.c
+++ b/src/bulkstat_unlink_test_modified.c
@@ -11,6 +11,8 @@
 #include <fcntl.h>
 #include <xfs/xfs.h>
 #include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
 
 int
 main(int argc, char *argv[])
diff --git a/src/iopat.c b/src/iopat.c
index f34de19..450d484 100644
--- a/src/iopat.c
+++ b/src/iopat.c
@@ -19,6 +19,9 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <xfs/xfs.h>
+#include <stdlib.h>
+#include <unistd.h>
+
 struct xfs_flock64 f;
 
 int main(int argc, char **argv)
diff --git a/src/unwritten_sync.c b/src/unwritten_sync.c
index bf61adf..cfe6f8d 100644
--- a/src/unwritten_sync.c
+++ b/src/unwritten_sync.c
@@ -3,6 +3,8 @@
 #include <errno.h>
 #include <malloc.h>
 #include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
 #include <xfs/xfs.h>
 
 /* test thanks to judith@sgi.com */
diff --git a/src/xfsctl.c b/src/xfsctl.c
index 3f82f4a..77d7603 100644
--- a/src/xfsctl.c
+++ b/src/xfsctl.c
@@ -16,7 +16,11 @@
  * Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <stdlib.h>
+#include <unistd.h>
 #include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
 
 #include <xfs/xfs.h>
 #include <xfs/jdm.h>
